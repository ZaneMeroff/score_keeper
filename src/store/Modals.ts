// Decorators
import { Action, Module, Mutation, VuexModule } from 'vuex-class-modules'

@Module
export default class Modals extends VuexModule {

  addPlayersModal: boolean = false
  clearDataModal: boolean = false
  deletePlayersModal: boolean = false
  resetScoreModal: boolean = false
  scoreLimitsModal: boolean = false

  get getShowAddPlayersModal(): boolean {
    return this.addPlayersModal
  }

  get getShowClearDataModal(): boolean {
    return this.clearDataModal
  }

  get getShowDeletePlayersModal(): boolean {
    return this.deletePlayersModal
  }

  get getShowResetScoreModal(): boolean {
    return this.resetScoreModal
  }

  get getShowScoreLimitsModal(): boolean {
    return this.scoreLimitsModal
  }

  // ------------------------------------------
  //       add players modal visibility
  // ------------------------------------------
  
  @Mutation
  addPlayersModalVisibility(show: boolean) {
    this.addPlayersModal = show
  }
  
  @Action
  action_addPlayersModalVisibility(show: boolean) {
    this.addPlayersModalVisibility(show)
  }

  // ------------------------------------------
  //        clear data modal visibility
  // ------------------------------------------
  
  @Mutation
  clearDataModalVisibility(show: boolean) {
    this.clearDataModal = show
  }
  
  @Action
  action_clearDataModalVisibility(show: boolean) {
    this.clearDataModalVisibility(show)
  }

  // ------------------------------------------
  //  confirm delete players modal visibility
  // ------------------------------------------
  
  @Mutation
  deletePlayersModalVisibility(show: boolean) {
    this.deletePlayersModal = show
  }

  @Action
  action_deletePlayersModalVisibility(show: boolean) {
    this.deletePlayersModalVisibility(show)
  }

  // ------------------------------------------
  //    confirm reset score modal visibility
  // ------------------------------------------
  
  @Mutation
  resetScoreModalVisibility(show: boolean) {
    this.resetScoreModal = show
  }

  @Action
  action_resetScoreModalVisibility(show: boolean) {
    this.resetScoreModalVisibility(show)
  }

  // ------------------------------------------
  //        score limit modal visibility
  // ------------------------------------------
  
  @Mutation
  scoreLimitModalVisibility(show: boolean) {
    this.scoreLimitsModal = show
  }

  @Action
  action_scoreLimitModalVisibility(show: boolean) {
    this.scoreLimitModalVisibility(show)
  }
}