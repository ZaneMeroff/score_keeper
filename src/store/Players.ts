// Vue & Decorators
import Vue from 'vue'
import { Action, Module, Mutation, VuexModule } from 'vuex-class-modules'

// Types
import { Player, PlayerData } from '@/types/players'

@Module
export default class Players extends VuexModule {

    playerData: PlayerData = {}

    get getPlayerData() {
      return this.playerData
    }

    // ------------------------------------------
    //              create player
    // ------------------------------------------

    @Mutation
    createPlayer(payload: Player) {
      Vue.set(this.playerData, payload.id, payload)
    }

    @Action
    action_createPlayer(payload: Player) {
      this.createPlayer(payload)
    }

    // ------------------------------------------
    //         modify target player score
    // ------------------------------------------

    @Mutation
    setPlayerScore(payload: { id: string, score: number }) {
      this.playerData[payload.id].score = payload.score
    }

    @Action
    action_setPlayerScore(payload: { id: string, score: number }) {
      this.setPlayerScore(payload)
    }

    // ------------------------------------------
    //         modify target player name
    // ------------------------------------------

    @Mutation
    setPlayerName(payload: { id: string, name: string }) {
      this.playerData[payload.id].name = payload.name
    }

    @Action
    action_setPlayerName(payload: { id: string, name: string }) {
      this.setPlayerName(payload)
    }

    // ------------------------------------------
    //        zero scores for all players
    // ------------------------------------------

    @Mutation
    zeroScores() {
      for (const player in this.playerData) {
        this.playerData[player].score = 0
      }
    }

    @Action
    action_zeroScores() {
      this.zeroScores()
    }
}