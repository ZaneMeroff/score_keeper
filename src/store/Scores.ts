// Decorators
import { Action, Module, Mutation, VuexModule } from 'vuex-class-modules'

// Types
import { Score, ScoreData } from '@/types/scores'
import Vue from 'vue'

@Module
export default class Scores extends VuexModule {

    scoreData: ScoreData = {}

    get getScoreData() {
      return this.scoreData
    }

    @Mutation
    setScoreData(payload: { id: string, updatedScore: number }) {
      // create or update score data based on id
      Vue.set(this.scoreData, payload.id, payload.updatedScore)
    }

    
}