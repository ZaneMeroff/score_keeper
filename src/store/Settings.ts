// Decorators
import Vue from 'vue'
import { Action, Module, Mutation, VuexModule } from 'vuex-class-modules'

@Module
export default class Settings extends VuexModule {

  darkMode: boolean = true
  minScore: number = 0
  maxScore: number = 10
  showScoreLimitsModal: boolean = false
  // showMenuDrawer: boolean = false

  get getDarkMode() {
    return this.darkMode
  }

  get getMinScore() {
    return this.minScore
  }

  get getMaxScore() {
    return this.maxScore
  }

  get getShowScoreLimitsModal() {
    return this.showScoreLimitsModal
  }

  // get getShowMenuDrawer() {
  //   return this.showMenuDrawer
  // }
  
  // ------------------------------------------
  //             toggle dark mode
  // ------------------------------------------

  @Mutation
  setDarkMode() {
    this.darkMode = !this.darkMode
  }

  @Action
  action_setDarkMode() {
    this.setDarkMode()
  }

  // ------------------------------------------
  //            set score limits
  // ------------------------------------------

  @Mutation
  setScoreLimits(payload: { min: number, max: number }) {
    this.minScore = payload.min
    this.maxScore = payload.max
  }

  @Action
  action_setScoreLimits(payload: { min: number, max: number }) {
    this.setScoreLimits(payload)
  }

  // ------------------------------------------
  //       score limit modal visibility
  // ------------------------------------------
  
  @Mutation
  setScoreLimitModalVisibility(show: boolean) {
    this.showScoreLimitsModal = show
  }

  @Action
  action_setScoreLimitModalVisibility(show: boolean) {
    this.setScoreLimitModalVisibility(show)
  }

  // ------------------------------------------
  //         menu drawer visibility
  // ------------------------------------------
  
  // @Mutation
  // setMenuDrawerlVisibility(show: boolean) {
  //   this.showMenuDrawer = show
  // }

  // @Action
  // action_setMenuDrawerlVisibility(show: boolean) {
  //   this.setMenuDrawerlVisibility(show)
  // }
}