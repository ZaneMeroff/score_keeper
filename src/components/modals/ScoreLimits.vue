<template>
  <div>
    
    <!-- score limits modal -->
    <q-dialog v-model="showModal" persistent>
      <q-card class="q-card" :dark="stateSettings.getDarkMode">

        <q-card-section>
          <!-- header -->
          <div class="text q-mb-sm" :style="stateSettings.getDarkModeText">Score Limits</div>

          <q-separator color="blue" inset></q-separator>

          <!-- min score number input -->
          <div class="q-mt-lg" style="color: #979797">min</div>
          <q-input  
            v-model.number="scoreMin" 
            style="font-size: 16px"
            type="number"
            :dark="stateSettings.getDarkMode"
            :rules="[ val => !!val.toString() || '* Required' ]"
            @input="onInputChange"
          />
          <!-- max score number input -->
          <div style="color: #979797">max</div>
          <q-input 
            v-model.number="scoreMax" 
            style="font-size: 16px"
            type="number"
            :dark="stateSettings.getDarkMode" 
            :rules="[ val => !!val.toString() || '* Required' ]"
            @input="onInputChange"
          />
          <q-btn-group flat class="q-btn-group">
            <!-- save button -->
            <q-btn
              flat
              class="q-btn"
              color="green"
              size="md"
              :label="'save'"
              :disable="disabled"
              @click="handleSaveBtn"
            />
            <!-- cancel button -->
            <q-btn
              flat
              class="q-btn"
              color="red"
              size="md"
              :label="'cancel'"
              @click="handleCancelBtn"
            />
          </q-btn-group>
        </q-card-section>
      </q-card>
    </q-dialog>

    <!-- error dialog -->
    <q-dialog v-model="showError" persistent>
      <q-card class="q-card" :dark="stateSettings.getDarkMode">
        <q-card-section>
          <div class="q-mb-md">
            <!-- header -->
            <div class="text q-mb-sm" :style="stateSettings.getDarkModeText"><q-icon color="red" name="error"></q-icon> Score Rules</div>
            <q-separator color="blue" inset></q-separator>
            <!-- rule text -->
            <div class="text q-my-md" :style="stateSettings.getDarkModeText">Min Score must be -500 to 0</div>
            <div class="text" :style="stateSettings.getDarkModeText">Max Score must be 1 to 500</div>
          </div>
          <!-- ok button -->
          <q-btn
            flat
            color="green"
            size="md"
            style="width: 100%"
            :label="'ok'"
            @click="() => {
              showError = false
              this.stateModals.action_scoreLimitModalVisibility(true)
            }"
          />
        </q-card-section>
      </q-card>
    </q-dialog>

  </div>
</template>

<script lang="ts">

// Decorators
import { Component, Prop, Vue, Watch } from 'vue-property-decorator'

// Vuex
import { stateModals, statePlayers, stateSettings } from '@/store/index'

@Component
export default class ScoreLimits extends Vue {

  @Prop(Boolean) private showModal!: boolean

  stateModals = stateModals
  statePlayers = statePlayers
  stateSettings = stateSettings

  scoreMin: number = 0
  scoreMax: number = 0

  showError: boolean = false
  disabled: boolean = false

  handleSaveBtn() {
    if (this.validateRules()) {
      // check in any players have a score outside of the new score limits
      // if so, reset all player scores to zero
      for (const player in this.statePlayers.getPlayerData) {
        let score = this.statePlayers.getPlayerData[player].score
        if (score > this.scoreMax || score < this.scoreMin) this.statePlayers.zeroScores()
      }
      // set new score limits and close modal
      this.stateSettings.action_setScoreLimits({ min: this.scoreMin, max: this.scoreMax })
      this.$emit('close')
    } else {
      this.showError = true
    }
    this.stateModals.action_scoreLimitModalVisibility(false)
  }

  handleCancelBtn() {
    this.setDefaults()
    this.stateModals.action_scoreLimitModalVisibility(false)
    this.$emit('close')
  }

  onInputChange() {
    if (!this.scoreMin.toString() || !this.scoreMax.toString()) this.disabled = true
    else this.disabled = false
  }

  setDefaults() {
    this.scoreMin = this.stateSettings.getMinScore
    this.scoreMax = this.stateSettings.getMaxScore
  }

  validateRules(): boolean {
    if ((this.scoreMin > 0 || this.scoreMin < -500) || (this.scoreMax > 500 || this.scoreMax < 1)) return false
    else return true
  }

  @Watch('showModal')
  setScoreLimits() {
    this.setDefaults()
  }
}
</script>

<style scoped>

  .q-card {
    width: 300px;
  }

  .text {
    color: #FFF;
    font-size: 16px;
    text-align: center;
  }

  .q-btn {
    width: 50%;
  }

  .q-btn-group {
    display: inline !important;
  }

  .q-item-section {
    display: flex;
    align-items: center;
    padding: 10px 0px 20px 0px;
  }

</style>
