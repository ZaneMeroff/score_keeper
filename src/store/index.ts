import Vue from 'vue'
import Vuex from 'vuex'
import VuexPersistence from 'vuex-persist'
import localforage from 'localforage'

import Modals from './Modals'
import Players from './Players'
import Settings from './Settings'

Vue.use(Vuex)

const vuexLocal = new VuexPersistence({
  storage: localforage as any,
  asyncStorage: true,
  modules: [ 'Modals', 'Players', 'Settings' ],
});

export const store: any = new Vuex.Store({
  // strict: process.env.DEV === 'true',
  state: {},
  modules: {},
  plugins: [ vuexLocal.plugin ],
})

export const stateModals= new Modals({ store, name: 'Modals' })
export const statePlayers= new Players({ store, name: 'Players' })
export const stateSettings = new Settings({ store, name: 'Settings' })
