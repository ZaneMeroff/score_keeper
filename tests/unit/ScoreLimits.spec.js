import { createLocalVue, shallowMount } from '@vue/test-utils'
import { stateModals, statePlayers, stateSettings } from '@/store/index'
import ScoreLimits from '@/components/modals/ScoreLimits.vue'
import Quasar, * as All from 'quasar'
import uuid from 'uuid/v4'
import Vuex from 'vuex'

// --------- Vuex & Quasar setup -------------

const localVue = createLocalVue()
localVue.use(Vuex, Quasar, {components: All, directives: All, plugins: All})

// ---------------- mocks --------------------

jest.mock('uuid/v4')
uuid.mockImplementation(() => '12345')

// ------------ create player ----------------

statePlayers.action_createPlayers(1)

// -------------------------------------------

describe('ScoreLimits', () => {

  describe('snapshots', () => {

    it('should render with default html', () => {
      const propsData = { showModal: true }
      const component = shallowMount(ScoreLimits, { localVue, propsData })
      
      expect(component.element).toMatchSnapshot()
    })
  })

  describe('props', () => {

    describe('showModal', () => {

      it('should accept a boolean', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        expect(component.vm.$props.showModal).toEqual(propsData.showModal)
      })
    })
  })

  describe('data', () => {

    describe('scoreMin', () => {

      it('should have a default value of 0', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        const expected = 0
        expect(component.vm.$data.scoreMin).toEqual(expected)
      })

      it('should accept a number', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        const newValue = -100

        component.setData({ scoreMin: newValue })

        expect(component.vm.$data.scoreMin).toEqual(newValue)
      })
    })

    describe('scoreMax', () => {

      it('should have a default value of 0', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        const expected = 0
        expect(component.vm.$data.scoreMax).toEqual(expected)
      })

      it('should accept a number', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        const newValue = 100

        component.setData({ scoreMax: newValue })

        expect(component.vm.$data.scoreMax).toEqual(newValue)
      })
    })

    describe('showError', () => {

      it('should have a default value of false', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        expect(component.vm.$data.showError).toEqual(false)
      })

      it('should accept a boolean', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        const newValue = true

        component.setData({ showError: newValue })

        expect(component.vm.$data.showError).toEqual(newValue)
      })
    })

    describe('disabled', () => {

      it('should have a default value of false', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        expect(component.vm.$data.disabled).toEqual(false)
      })

      it('should accept a boolean', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        const newValue = true

        component.setData({ disabled: newValue })
        expect(component.vm.$data.disabled).toEqual(newValue)
      })
    })
  })

  describe('methods', () => {

    describe('handleSaveBtn', () => {

      it('should zero scores if player has score outside of new limits', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        statePlayers.action_setPlayerScore({ id: '12345', score: 5 })
        expect(statePlayers.getPlayerData['12345'].score).toEqual(5)

        component.setData({ scoreMax: 2 })
        component.vm.handleSaveBtn()

        expect(statePlayers.getPlayerData['12345'].score).toEqual(0)

        // teardown
        stateSettings.action_setScoreLimits({ min: 0, max: 10 })
      })

      it('should set new score limits if validateRules returns true', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        expect(stateSettings.getMinScore).toEqual(0)
        expect(stateSettings.getMaxScore).toEqual(10)

        component.setData({ scoreMin: -5, scoreMax: 5 })
        component.vm.handleSaveBtn()

        expect(stateSettings.getMinScore).toEqual(-5)
        expect(stateSettings.getMaxScore).toEqual(5)

        // teardown
        stateSettings.action_setScoreLimits({ min: 0, max: 10 })
      })

      it('should set scoreLimitsModal to false if validateRules returns true', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        stateModals.action_scoreLimitModalVisibility(true)
        expect(stateModals.getShowScoreLimitsModal).toEqual(true)

        component.vm.handleSaveBtn()

        expect(stateModals.getShowScoreLimitsModal).toEqual(false)
      })

      it('should emit close event if validateRules return true', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        component.setData({ scoreMin: 0 })
        component.setData({ scoreMax: 10 })
        component.vm.handleSaveBtn()

        const expected = { 'close': [[]] }
        expect(component.emitted()).toEqual(expected)
      })

      it('should set showError=true if validateRuels returns false', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        component.vm.handleSaveBtn()

        expect(component.vm.validateRules()).toEqual(false)
        expect(component.vm.$data.showError).toEqual(true)
      })
    })

    describe('handleCancelBtn', () => {
      
      it('should call setDefaults', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        const spy = jest.spyOn(component.vm, 'setDefaults')

        component.vm.handleCancelBtn()

        expect(spy).toHaveBeenCalledTimes(1)
      })

      it('should set scoreLimitsModal to false', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        stateModals.action_scoreLimitModalVisibility(true)
        expect(stateModals.getShowScoreLimitsModal).toEqual(true)

        component.vm.handleCancelBtn()

        expect(stateModals.getShowScoreLimitsModal).toEqual(false)
      })

      it('should emit close event', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        component.vm.handleCancelBtn()

        const expected = { 'close': [[]] }
        expect(component.emitted()).toEqual(expected)
      })
    })

    describe('onInputChange', () => {
      
      it('should set disabled to true if scoreMin is empty', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })

        component.setData({ scoreMin: '' })
        component.vm.onInputChange()

        expect(component.vm.$data.disabled).toEqual(true)
      })

      it('should set disabled to true if scoreMax is empty', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })

        component.setData({ scoreMax: '' })
        component.vm.onInputChange()

        expect(component.vm.$data.disabled).toEqual(true)
      })

      it('should set disabled to false if scoreMin & scoreMax have values', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })

        component.setData({ scoreMin: -100 })
        component.setData({ scoreMax: 100 })
        component.vm.onInputChange()

        expect(component.vm.$data.disabled).toEqual(false)
      })
    })

    describe('setDefaults', () => {
      
      it('should set scoreMin & scoreMax', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        component.vm.setDefaults()

        const expectedScoreMin = 0
        const expectedScoreMax = 10
        expect(component.vm.$data.scoreMin).toEqual(expectedScoreMin)
        expect(component.vm.$data.scoreMax).toEqual(expectedScoreMax)
      })
    })

    describe('validateRules', () => {
      
      it('should return false if scoreMin > 0', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })

        component.setData({ scoreMin: 1 })

        expect( component.vm.validateRules()).toEqual(false)
      })

      it('should return false if scoreMin < -500', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })

        component.setData({ scoreMin: -501 })

        expect( component.vm.validateRules()).toEqual(false)
      })

      it('should return false if scoreMax < 1', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })

        component.setData({ scoreMax: 0 })

        expect( component.vm.validateRules()).toEqual(false)
      })

      it('should return false if scoreMax > 500', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })

        component.setData({ scoreMax: 501 })

        expect( component.vm.validateRules()).toEqual(false)
      })

      it('should return true if score limits are correct', () => {
        const propsData = { showModal: true }
        const component = shallowMount(ScoreLimits, { localVue, propsData })

        component.setData({ scoreMin: -100})
        component.setData({ scoreMax: 100 })

        expect( component.vm.validateRules()).toEqual(true)
      })
    })
  })

  describe('watch', () => {

    describe('showModal', () => {

      it('should call setDefaults if showModal changes', async () => {
        const propsData = { showModal: true }
        const spy = jest.spyOn(ScoreLimits.options.methods, 'setDefaults')
        const component = shallowMount(ScoreLimits, { localVue, propsData })
        
        await component.setProps({ showModal: false})

        expect(spy).toHaveBeenCalled()
      })
    })
  })
})
