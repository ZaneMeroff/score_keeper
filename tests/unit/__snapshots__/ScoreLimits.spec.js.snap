// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreLimits snapshots should render with default html 1`] = `
<div>
  <q-dialog-stub
    persistent="true"
    position="standard"
    value="true"
  >
    <q-card-stub
      class="q-card"
      dark="true"
      tag="div"
    >
      <q-card-section-stub
        tag="div"
      >
        <div
          class="text q-mb-sm"
          style="color: rgb(255, 255, 255);"
        >
          Score Limits
        </div>
         
        <q-separator-stub
          color="blue"
          inset="true"
        />
         
        <div
          class="q-mt-lg"
          style="color: rgb(151, 151, 151);"
        >
          min
        </div>
         
        <q-input-stub
          dark="true"
          rules="[Function]"
          style="font-size: 16px;"
          type="number"
          value="0"
        />
         
        <div
          style="color: rgb(151, 151, 151);"
        >
          max
        </div>
         
        <q-input-stub
          dark="true"
          rules="[Function]"
          style="font-size: 16px;"
          type="number"
          value="0"
        />
         
        <q-btn-group-stub
          class="q-btn-group"
          flat="true"
        >
          <q-btn-stub
            align="center"
            class="q-btn"
            color="green"
            flat="true"
            label="save"
            ripple="true"
            size="md"
          />
           
          <q-btn-stub
            align="center"
            class="q-btn"
            color="red"
            flat="true"
            label="cancel"
            ripple="true"
            size="md"
          />
        </q-btn-group-stub>
      </q-card-section-stub>
    </q-card-stub>
  </q-dialog-stub>
   
  <q-dialog-stub
    persistent="true"
    position="standard"
  >
    <q-card-stub
      class="q-card"
      dark="true"
      tag="div"
    >
      <q-card-section-stub
        tag="div"
      >
        <div
          class="q-mb-md"
        >
          <div
            class="text q-mb-sm"
            style="color: rgb(255, 255, 255);"
          >
            <q-icon-stub
              color="red"
              name="error"
              tag="i"
            />
             Score Rules
          </div>
           
          <q-separator-stub
            color="blue"
            inset="true"
          />
           
          <div
            class="text q-my-md"
            style="color: rgb(255, 255, 255);"
          >
            Min Score must be -500 to 0
          </div>
           
          <div
            class="text"
            style="color: rgb(255, 255, 255);"
          >
            Max Score must be 1 to 500
          </div>
        </div>
         
        <q-btn-stub
          align="center"
          color="green"
          flat="true"
          label="ok"
          ripple="true"
          size="md"
          style="width: 100%;"
        />
      </q-card-section-stub>
    </q-card-stub>
  </q-dialog-stub>
</div>
`;
